### ISSUE TITLE
Package request: Smalltalk

### ISSUE THREAD

**Issue by s-c-p** [termux/termux-packages#3196]
For an introduction to OOP, I see Smalltalk as a better alternative to Java, especially when teaching younger students.  
The problem with trying to teach OOP with Python, C++, etc. is that students often mis-learn that simply grouping a bunch of related functions under a class name qualifies as OOP. And despite course coverage they never *utilize* concepts like SOLID, contracts, etc..

**Package description**
This package should allow user to do all things `gst` (GNU Smalltalk) does in any POSIX-y OS.

**Link to home page and sources**
1. Home page: http://smalltalk.gnu.org/download
2. Source code: http://git.sv.gnu.org/r/smalltalk.git

**Additional information**
I tried comprehending the README but I can't compile (as in, lack knowledge and experience). I found [this issue](https://github.com/termux/termux-packages/issues/2977) about a JVM port (?) of Smalltalk called redline, but I created a new issue to separate concerns -- the end result we want is Smalltalk on Android.

**Comment by yubrajbhoi** [comment]
I would love to have gst on Termux. I tried compiling it a few days ago but got stuck in a GnuTLS error.

**Comment by Grimler91** [comment]
@yubrajbhoi that probably means that you didn't have libgnutls-dev or libgnutls installed. It built fine for me when I tried it

**Comment by yubrajbhoi** [comment]
@Grimler91 I did installed those dev packages. Can you upload your build log and the package ? I would really appreciate it.

**Comment by Grimler91** [comment]
@yubrajbhoi I had to apply these patches:
```diff
diff --git a/packages/i18n/i18n.c b/packages/i18n/i18n.c
index fb050f83..77e3a50d 100644
--- a/packages/i18n/i18n.c
+++ b/packages/i18n/i18n.c
@@ -57,7 +57,11 @@
 #include <limits.h>
 #include <stdlib.h>
 #include <string.h>
+#if defined(__ANDROID__)
+#include <libandroid-support/langinfo.h>
+#else
 #include <langinfo.h>
+#endif
 #include <errno.h>
 
 #if defined(_WIN32) || defined(__CYGWIN32__) || defined(__CYGWIN__) || defined(Win32) || defined(__WIN32)
diff --git a/lib-src/getdtablesize.c b/lib-src/getdtablesize.c
index 417f127f..a4764235 100644
--- a/lib-src/getdtablesize.c
+++ b/lib-src/getdtablesize.c
@@ -78,15 +78,6 @@ getdtablesize(void)
   int tableSize = -1;
 #endif
 
-#ifdef RLIMIT_NOFILE           /* who is non POSIX but has this? */
-  struct rlimit res;
-  if (tableSize >= 0)
-    return tableSize;
-
-  if (getrlimit(RLIMIT_NOFILE, &rlimit) != -1)
-    tableSize = rlimit.rlim_max;
-#endif
-
   if (tableSize >= 0)
     return tableSize;
 
```
and then configured with `./configure --disable-generational-gc --prefix=$PREFIX`. And then `make` and `make install`.
If this doesn't work for you then please share the error you get. 

I can probably package it and provide debs in a few days as well

**Comment by yubrajbhoi** [comment]
@Grimler91 Thanks for the patch. I will try to build it again.

**Comment by yubrajbhoi** [comment]
@Grimler91 Unfortunately it failed with the same error

```
smalltalk-3.2.5/packages/net/gnutls-wrapper.c:128: undefined reference to `gnutls_global_init'
```

**Comment by Grimler91** [comment]
@yubrajbhoi what's your output of `termux-info`?

**Comment by yubrajbhoi** [comment]
@Grimler91 Here it is
```
Updatable packages:
radare2/stable 3.2.0 arm [upgradable from: 3.1.3]
System information:
Linux localhost 3.18.71-perf-gea366bb #1 SMP PREEMPT Wed Oct 17 04:24:20 CDT 2018 armv7l Android
Termux-packages arch:
arm
Android version:
8.1.0
Device manufacturer:
motorola
Device model:
Moto G (5)
```

**Comment by Grimler91** [comment]
Do you have pkg-config installed? (try installing that otherwise)

If that doesn't work then this might be a arm-only error. Might work to add -lgnutls to a suitable flag in a suitable Makefile. 

I can try it on my own device tonight.

**Comment by yubrajbhoi** [comment]
@Grimler91 Yes I have pkg-config installed. I will try to edit some Makefile.

**Comment by yubrajbhoi** [comment]
@Grimler91 Wow! I managed to build it. I had to edit `smalltalk-3.2.5/packages/net/Makefile` and add `-lgnutls` to `CFLAGS`. Many thanks for your help.

**Comment by s-c-p** [comment]
Can we make a Pull Request that enables us to just `apt install smalltalk` in normal (i.e. devices with no root or dev libraries) Termux installation, instead of compiling all the stuff.

**Comment by s-c-p** [comment]
Can we make a Pull Request that enables us to just `apt install smalltalk` in normal (i.e. devices with no root or dev libraries) Termux installation, instead of compiling all the stuff.
